contrib/endpoints/src/api_manager/api_manager_impl.cc

```diff
@@ -75,11 +85,41 @@ utils::Status ApiManagerImpl::Init() {
     global_context_->cloud_trace_aggregator()->Init();
   }
 
-  for (auto it : service_context_map_) {
-    if (it.second->service_control()) {
-      it.second->service_control()->Init();
-    }
+  if (service_context_map_.size() > 0) {
+    // Already initialized
+    return utils::Status::OK;
   }
+
+  config_manager_.reset(new ConfigManager(
+      global_context_,
+      [this](const utils::Status &status,
+             const std::vector<std::pair<std::string, int>> &configs) {
+        if (status.ok()) {
+          std::vector<std::pair<std::string, int>> rollouts;
+          std::string config_id;
+
+          for (auto item : configs) {
+            if (AddConfig(item.first, config_id)) {
+              rollouts.push_back({config_id, item.second});
+            } else {
+              initialization_status_ =
+                  utils::Status(Code::ABORTED, "Invalid service config");
+              return;
+            }
+          }
+
+          for (auto it : service_context_map_) {
```

Comment:
not to call all service_contexts.  only ones created by this call.   It is better to call it at line 103